Userland/Libraries/LibELF/DynamicLinker.cpp

```diff
@@ -45,7 +45,7 @@ namespace {
 ByteString s_main_program_path;
 
 // The order of objects here corresponds to the "load order" from POSIX specification.
-OrderedHashMap<ByteString, NonnullRefPtr<ELF::DynamicObject>> s_global_objects;
+OrderedHashMap<ByteString, NonnullOwnPtr<ELF::DynamicObject>> s_global_objects;
```

Comment:
Not sure bout this one. If we need fancy ref-coutning for Dynamic Object, the proper way would be to implement that custom ref-counting on the object directly, rather than inheriting from RefCounted. RefPtr doesn't require you to use RefCounted, simply that the object has the required ref/unref calls.

Reply:
I found it easier to reason about the DynamicObjects' lifetimes when there's a single owner and two separate refcounts that aren't obligatory.

Reply:
Looking into the next commit a bit, I'm not sure what the difference between your proposed whacky ownership model and the "normal" RefPtr model; `dlopen_ref` and `dlopen_unref` are just `ref` and `unref`, your manual `unref` call is exactly equivalent to setting the pointer to null. your manual `ref` calls can stay as they are.